### PR TITLE
StatusBar colouring, tag support, and oUF checking

### PR DESCRIPTION
**Pull request by sbaildon**
Small PR to:

* Add support for colouring the widget
* Tag support
* Don't waste time loading if oUF isn't installed

Doesn't remove any features, will stay compatible with current installs.
Code for the tags is a bit repetitive, and I'm not convinced with the naming scheme for tags. Any ideas?

### CODE REVIEW

File: oUF_ArtifactPower.lua [hunk 1]
```diff
@@ -146,6 +147,10 @@ local function Enable(self, unit)
 		element:SetStatusBarColor(.901, .8, .601)
 	end
 
+	if element:IsObjectType('StatusBar') and element.colorArtifactQuality then
+		element:SetStatusBarColor(GetItemQualityColor(6))
+	end
+
```
Comment (Rainrider) [inline oUF_ArtifactPower.lua]: I'm against that as it is very limiting. Some refactoring in the lines above it solve the lack of options for coloring. I'll fix this myself, thanks for pointing out the use case.

File: oUF_ArtifactPower.lua [hunk 2]
```diff
@@ -46,6 +46,7 @@ are not set by the layout.
 
 local _, ns = ...
 local oUF = ns.oUF or oUF
+assert(oUF, "oUF ArtifactPower was unable to locate oUF install")
```
Comment (Rainrider) [inline oUF_ArtifactPower.lua]: This is not needed. oUF is a required dependency (see the .toc)

**Review by Rainrider — CHANGES_REQUESTED** [review]
Thank you for the PR! Could you please resolve the requested changes, I'll merge then. Basically just leave it to the tags and remove the rest

**Comment by sbaildon** [comment]
Rebased to to remove the changes you're not interested in, and to squash the various tag support commits.

What's the plan for supporting colour? I figured that people would be expected to call `SetStatusBarColor` themselves when they initialise the `StatusBar`

**Comment by Rainrider** [comment]
Yes, that is the plan, I'll just factor the color check in its own if block. I'll review the rebase when I'm home later, thank you again

**Pull request merged**